ci/scripts/min-deps.py

```diff
@@ -0,0 +1,102 @@
+#!python3
+from __future__ import annotations
+
+import argparse
+import sys
+from collections import deque
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib
+
+from packaging.requirements import Requirement
+from packaging.version import Version
+
+if TYPE_CHECKING:
+    from collections.abc import Generator, Iterable
+
+
+def min_dep(req: Requirement) -> Requirement:
+    """
+    Given a requirement, return the minimum version specifier.
+
+    Example
+    -------
+
+    >>> min_dep(Requirement("numpy>=1.0"))
+    "numpy==1.0"
+    """
+    req_name = req.name
+    if req.extras:
+        req_name = f"{req_name}[{','.join(req.extras)}]"
+
+    # TODO: Should this be allowed?
+    if not req.specifier:
+        return Requirement(req_name)
+
+    min_version = Version("0.0.0.a1")
+    for spec in req.specifier:
+        if spec.operator in [">", ">=", "~="]:
+            min_version = max(min_version, Version(spec.version))
+        elif spec.operator == "==":
+            min_version = Version(spec.version)
+
+    # TODO: should this return `~=` or `==`?
```

Comment:
both should be identical with the last item being `.*`. since `~=` has unintuitive semantics, I’d prefer not to use it.